ethcore/sync/src/chain/mod.rs

```diff
@@ -762,22 +762,20 @@ impl ChainSync {
 						}
 					}
 
-					// Only ask for old blocks if the peer has a higher difficulty than the last imported old block
-					let last_imported_old_block_difficulty = self.old_blocks.as_mut().and_then(|d| {
-						io.chain().block_total_difficulty(BlockId::Number(d.last_imported_block_number()))
-					});
+					// Only ask for old blocks if the peer has an equal or higher difficulty
+					let equal_or_higher_difficulty = peer_difficulty.map_or(false, |pd| pd >= syncing_difficulty);
```

Comment:
Post fork block we were getting some dodgy blocks from some nodes, so tightening this back up. Still addresses https://github.com/paritytech/parity-ethereum/issues/9225